sdk/cosmosdb/cosmos/src/queryExecutionContext/EndpointComponent/OffsetLimitEndpointComponent.ts

```diff
@@ -3,15 +3,39 @@
 import type { DiagnosticNodeInternal } from "../../diagnostics/DiagnosticNodeInternal.js";
 import type { Response } from "../../request/index.js";
 import type { ExecutionContext } from "../ExecutionContext.js";
+import type { FeedOptions } from "../../request/index.js";
+import type { ContinuationTokenManager } from "../ContinuationTokenManager.js";
 import { getInitialHeader, mergeHeaders } from "../headerUtils.js";
 
 /** @hidden */
 export class OffsetLimitEndpointComponent implements ExecutionContext {
+  private continuationTokenManager: ContinuationTokenManager | undefined;
+
   constructor(
     private executionContext: ExecutionContext,
     private offset: number,
     private limit: number,
-  ) {}
+    options?: FeedOptions,
+  ) {
+    // Get the continuation token manager from options if available
+    this.continuationTokenManager = (options as any)?.continuationTokenManager;
+
+    // Check continuation token for offset/limit values during initialization
+    if (options?.continuationToken) {
+      try {
+        const parsedToken = JSON.parse(options.continuationToken);    
+        // Use continuation token values if available, otherwise keep provided values
+        if (parsedToken.offset) {
+          this.offset = parsedToken.offset;
+        }
+        if (parsedToken.limit) {
+          this.limit = parsedToken.limit;
+        }
+      } catch {
+        // If parsing fails, use the provided offset/limit values from query plan
```

Comment:
IMO, Failing the request is better and more predictable path here. 
We should not continue with some assumption which is not transparent to user